src/components/location-detail/location-services.tsx

```diff
@@ -77,16 +78,32 @@ export default function LocationServices({
   const previousCategory =
     ROUTE_TO_CATEGORY_MAP[previousParams?.params.route as string];
   const previousRoute = usePreviousRoute();
+  console.log("previousRoute", previousRoute, previousParams);
+  const previousSubcategory =
+    previousParams?.params.locationSlugOrPersonalCareSubCategory;
+
+  const previousSubcategoryAsCategory =
+    previousSubcategory === "mental-health" ||
+    previousSubcategory === "legal-services" ||
+    previousSubcategory === "employment"
+      ? (previousSubcategory as Category)
+      : undefined;
```

Comment:
@shakilhossain1 Would the logic be simpler if we conditionally override previousSubcategory, rather than creating a new variable? Like this:


```suggestion
previousCategory =
    previousSubcategory === "mental-health" ||
    previousSubcategory === "legal-services" ||
    previousSubcategory === "employment"
      ? (previousSubcategory as Category)
      : previousCategory;
```

Reply:
@codex Is there a simpler way to write this?

Reply:
Yes. I simplified the logic here.